lib/link_checker.rb

```diff
@@ -0,0 +1,79 @@
+require "uri"
+
+module LinkChecker
+module_function
+
+  def gather_urls(yaml_node)
+    urls = Set[]
+    if yaml_node.class == String
+      urls.merge(Set.new(URI.extract(yaml_node, %w[http https]).to_set))
```

Comment:
Is the `Set.new` redundant as `to_set` is being called on `URI.extract(yaml_node, %w[http https])`?